File: samples/server/petstore/springboot-delegate-no-response-entity/src/main/java/org/openapitools/api/ExampleApiController.java
package org.openapitools.api;

import org.openapitools.model.ExampleResponse;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.CookieValue;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RequestPart;
import org.springframework.web.multipart.MultipartFile;

import javax.validation.constraints.*;
import javax.validation.Valid;

import java.util.List;
import java.util.Map;
import java.util.Optional;
import javax.annotation.Generated;

@Generated(value = "org.openapitools.codegen.languages.SpringCodegen")
@RestController
@RequestMapping("${openapi.dummy.base-path:}")
public class ExampleApiController implements ExampleApi {

    private final ExampleApiDelegate delegate;

    public ExampleApiController(@Autowired(required = false) ExampleApiDelegate delegate) {
        this.delegate = Optional.ofNullable(delegate).orElse(new ExampleApiDelegate() {});
    }

    @Override
    public ExampleApiDelegate getDelegate() {
        return delegate;
    }

}

File: samples/server/petstore/kotlin-springboot-delegate/src/main/kotlin/org/openapitools/api/ExampleApiController.kt
package org.openapitools.api

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import java.util.Optional

@jakarta.annotation.Generated(value = ["org.openapitools.codegen.languages.KotlinSpringServerCodegen"])
@Controller
@RequestMapping("\${openapi..base-path:}")
class ExampleApiController(
        @org.springframework.beans.factory.annotation.Autowired(required = false) delegate: ExampleApiDelegate?
) : ExampleApi {
    private val delegate: ExampleApiDelegate

    init {
        this.delegate = Optional.ofNullable(delegate).orElse(object : ExampleApiDelegate {})
    }

    override fun getDelegate(): ExampleApiDelegate = delegate
}
